### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by RahullRawat**
CSB Link - https://nnvrfv.csb.app/

- Added login ,logout and signup functionality

**Login credentials :-** 
Email - johndoe@gmail.com
Password - johnDoe123

### CODE REVIEW

File: src/context/AuthContext.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+				});
+				navigate("/");
+			}
+		} catch (err) {
+			authDispatch({ type: "ERROR" });
+		}
+	};
+
+	const signOut = () => {
+		authDispatch({ type: "TOKEN-REMOVED" });
```
Comment (VishalPatil18) [inline src/context/AuthContext.js]: This type could be named better. As we know that we are removing the token on signout but it doesn't make it clear in this dispatch. You could name it something as `LOGOUT`

File: src/context/AuthContext.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+	return (
+		<AuthContext.Provider
+			value={{
+				firstName,
+				email,
+				password,
+				userData,
+				token,
+				error,
```
Comment (VishalPatil18) [inline src/context/AuthContext.js]: You could add these in `authState` and pass only the `authState` as value to the provider.

**Review by VishalPatil18 — APPROVED** [review]
Just a couple of suggestions. The rest looks quite good. Great work @RahullRawat 👍

**Review by AtulPant2704 — APPROVED** [review]
- Great naming convention and very well formatted data made easy to review.
- Destructing objects was also good.

**Pull request merged**